apps/store/src/components/PriceCalculator/priceCalculatorAtoms.ts

```diff
@@ -0,0 +1,117 @@
+import { datadogLogs } from '@datadog/browser-logs'
+import { type Atom, atom, type Getter, useAtomValue, useStore } from 'jotai'
+import { atomFamily } from 'jotai/utils'
+import { useEffect } from 'react'
+import { priceTemplateAtom } from '@/components/ProductPage/PurchaseForm/priceTemplateAtom'
+import {
+  type PriceIntentFragment,
+  type ShopSessionCustomerFragment,
+} from '@/services/graphql/generated'
+import { setupForm } from '@/services/PriceCalculator/PriceCalculator.helpers'
+import { type Form } from '@/services/PriceCalculator/PriceCalculator.types'
+import { useShopSession } from '@/services/shopSession/ShopSessionContext'
+
+export const currentPriceIntentIdAtom = atom<string | null>(null)
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const priceIntentAtomFamily = atomFamily((priceIntentId: unknown) =>
+  atom<PriceIntentFragment | null>(null),
+)
+
+export const priceIntentAtom = atom<PriceIntentFragment>((get) => {
+  const priceIntentId = getAtomValueOrThrow(get, currentPriceIntentIdAtom)
+  const value = get(priceIntentAtomFamily(priceIntentId))
+  if (value == null) {
+    throw new Error('priceIntentAtom must have value')
+  }
```

Comment:
Would be possible to do the following?

```suggestion
  const value = getAtomValueOrThrow(get, priceIntentAtomFamily(priceIntentId))
```